sdk/src/beta9/cli/worker.py

```diff
@@ -207,3 +207,105 @@ def drain_worker(
             terminal.warn(text)
         else:
             terminal.success(f"Worker {worker_id} has been drained.")
+
+
+@click.group(
+    name="machine",
+    help="Manage machines.",
+    cls=ClickManagementGroup,
+)
+def machine():
+    pass
+
+
+@machine.command(
+    name="drain",
+    help="""
+    Drain all workers on a specific machine.
+
+    This command will find all workers running on the specified machine ID and drain them.
+    When a worker is drained, all running containers on it will be stopped. The worker will 
+    continue until its idle timeout if was cordoned before being drained. 
+    """,
+    epilog="""
+      Examples:
+
+        # Drain all workers on a specific machine
+        {cli_name} machine drain 0d123123
+    """,
+)
+@click.argument(
+    "machine_id",
+    nargs=1,
+    required=True,
+)
+@extraclick.pass_service_client
+def drain_machine(
+    service: ServiceClient,
+    machine_id: str,
+):
+    res = service.gateway.list_workers(ListWorkersRequest())
+    if not res.ok:
+        return terminal.error(f"Failed to list workers: {res.err_msg}")
+
+    matching_workers = [w.id for w in res.workers if w.machine_id == machine_id]
+
+    if not matching_workers:
+        return terminal.error(f"No workers found for machine ID: {machine_id}")
+
+    terminal.info(f"Found {len(matching_workers)} workers on machine {machine_id}")
+
+    for worker_id in matching_workers:
+        res = service.gateway.drain_worker(DrainWorkerRequest(worker_id=worker_id))
+        if not res.ok:
+            text = res.err_msg.capitalize()
+            terminal.warn(text)
+        else:
+            terminal.success(f"Worker {worker_id} has been drained.")
```

Comment:
I'm not sure if we can invoke across modules, but can we try it so we don't have duplicate logic?

See how to invoke in Click here.
https://github.com/beam-cloud/beta9/blob/main/sdk/src/beta9/cli/deployment.py#L69

Reply:
Oh hm, you may want `ctx.invoked_subcommand(...)`